app/src/main/java/com/github/bgabriel998/softwaredevproject/general/SettingsActivity.java

```diff
@@ -1,20 +1,164 @@
+                            rangeChanged();
+                            break;
+                        case "language_preference":
+                            languageChanged((String) prefs.getAll().get("language_preference"));
+                            break;
+                        case "disable_caching":
+                            disableCachingChanged();
+                            break;
+                        case "offline_mode_key":
+                            offlineModeChanged();
+                    }
+                }
+    };
+
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        setContentView(R.layout.activity_settings);
-        ToolbarHandler.SetupToolbar(this, TOOLBAR_TITLE);
+        setContentView(R.layout.settings_activity);
+        Intent intent = getIntent();
+        ToolbarHandler.SetupToolbarCustom(this, this,TOOLBAR_TITLE);
+
+        if (savedInstanceState == null) {
+            getSupportFragmentManager()
+                    .beginTransaction()
+                    .replace(R.id.settings, new SettingsFragment())
+                    .commit();
+        }
+        ActionBar actionBar = getSupportActionBar();
+        if (actionBar != null) {
+            actionBar.setDisplayHomeAsUpEnabled(true);
+        }
+
+        SharedPreferences sharedPref = PreferenceManager.getDefaultSharedPreferences(this);
+        sharedPref.registerOnSharedPreferenceChangeListener(listener);
+    }
+
+    /**
+     * Needed static class to implement the default android preferences
+     */
+    public static class SettingsFragment extends PreferenceFragmentCompat {
+        @Override
+        public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
+            setPreferencesFromResource(R.xml.root_preferences, rootKey);
+        }
+    }
+
+    /**
+     * Change measurement system
+     */
+    private void measurementSystemChanged(){
+        //TODO : Implement
+        Toast.makeText(this,"Setting not implemented yet !", Toast.LENGTH_SHORT).show();
+    }
+
+    /**
+     * Change measurement system
+     */
+    private void rangeChanged(){
+        //TODO : Implement
+        Toast.makeText(this,"Setting not implemented yet !", Toast.LENGTH_SHORT).show();
+    }
+
+    /**
+     * Change language
+     */
+    private void languageChanged(String value){
+        Locale locale = new Locale("en");
+        switch (value){
+            case "german":
+                setLocale("de");
+                break;
+            case "french":
+                setLocale("fr");
+                break;
+            case "english":
+                setLocale("en");
+                break;
+            case "swedish":
+                setLocale("sv");
+                break;
+        }
+
+    }
+
+    /**
+     * Change application language
+     * @param lang language code
+     */
+    public void setLocale(String lang) {
```

Comment:
It's quite hard to understand what happens inside this method. Maybe it would be better to add some more comments?